modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java

```diff
@@ -443,6 +453,8 @@ private void performAddWriteCommittedWithCleanup(
 
             assert result.status() == AddWriteCommittedResultStatus.SUCCESS : "rowId=" + rowId + ", result=" + result;
         }
+
+        modificationCounter.updateValue(1, commitTs);
```

Comment:
I think it's not really efficient to update counter for every row during batch processing, those I would propose to move this on a few levels up on the call stack: to the `else` brach of `if (trackWriteIntent)` in `handleUpdate`, and to similar place in `processEntriesUntilBatchLimit`. WDYT?